h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -170,6 +170,16 @@ private static int ensureBuilderIndex(String urlName) {
     return mb;
   }
 
+  /**
+   * Factory method to create a ModelBuilder instance from a clone of a given {@code parms} instance of Model.Parameters.
+   */
+  public static <B extends ModelBuilder, MP extends Model.Parameters> B make(MP parms) {
+    Key<Model> mKey = ModelBuilder.defaultKey(parms.algoName());
+    Job<Model> mJob = new Job<>(mKey, parms.javaName(), parms.algoName());
+    B newMB = ModelBuilder.make(parms.algoName(), mJob, mKey);
+    newMB._parms = parms.clone();
+    return newMB;
+  }
```

Comment:
looks good to me

Reply:
the method looks fine, it deserves a simple unit test in h2o-core that will:

1. check that created instance is of expected type
2. check that _job, and _result are not null
3. check that parameters a cloned, not the same instance

Reply:
@michalkurka Yes, I was thinking about this kind of test that you described. 
One of the reasons why I didn't write tests in h2o-core at once was that we need algos or concrete classes that implements Model.Parameters abstract class( which also live in h2o-algos module).
In `ModelBuilderTest` we have `DummyModelBuilder` and `DummyModelParameters` and everything is dummy. This is maybe fine for a non-static functionality but in my case I need to test `ModelBuilder.make()` static method, which in addition uses static field of `ModelBuilder` -> ALGOBASES which is empty if we don't do registration of algos. 

All these ^^ makes it quite difficult to test.

Also I don't feel like those 3  checks guarantee that the ModelBuilder could be successfully trained. For example _job, and _result are not null even for ModelBuilder that was already trained once. Maybe checking for job ids would be needed. 

Is it convenient to register algos from within the test suite?

Reply:
@michalkurka I have found a solution - registered DummyModelBuilder through the side effect of the existing method/constructor. This is similar to how we register algos in `RegisterAlgos.java`. Let me know if you have another ideas.